bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/TeaserImpl.java

```diff
@@ -44,10 +46,30 @@
     @ScriptVariable
     protected Page currentPage;
 
+    /**
+     * The current component.
+     */
+    @ScriptVariable
+    private Component component;
```

Comment:
instead of this, I would make the `component` field in `v1/TeaserImpl` `protected` so that it can be used in v2.

Reply:
When making the component field protected in v1/TeaserImpl, I'm getting this error: 'Field TeaserImpl.component masks field in superclass com.adobe.cq.wcm.core.components.util.AbstractComponentImpl [com.adobe.cq.wcm.core.components.internal.models.v1.TeaserImpl] In TeaserImpl.java MF_CLASS_MASKS_FIELD'
Do you think is better to let the implementation as it is now, or should I let the component field private and create a getter for it?

Reply:
ah ok, we can then leave it as it is in the PR